ProcessMaker/Helpers/ArrayHelper.php

```diff
@@ -39,4 +40,35 @@ public static function getArrayDifferencesWithFormat(array $changedArray, array
         $arrayDiff = $arrayHelper->formatChanges($displayChanges, $displayOriginal);
         return $arrayDiff;
     }
+
+    /**
+     * This method swaps groups of id numbers "1,2,3" for specific column names values "'A','B','C'" 
+     * using the Tables information through Models
+     * The method returns a String value
+     * The packageName parameter is optional, only use with Packages. Default value Null
+     * @param string $modelName
+     * @param string $ids
+     * @param string $columnName
+     * @param string $packageName
+     * @return string
+     */
+    public static function getNamesByIds(string $modelName, string $ids, string $columnName, string $packageName = null): string
```

Comment:
Please this ProcessMaker/Helpers/ArrayHelper.php is related to the function arrays I think so this function needs to transform array in string like 

```
public static function convertArrayToString($array, $concat = ', ')
{
       return 'A, B, C';
}
```

please, move the logic with queries into other class related to that

Reply:
solved. A complete Refactor was done.